tests/ord-aggregator/tests/handler_test.go

```diff
@@ -0,0 +1,254 @@
+	expectedNumberOfProducts        = 1
+	expectedNumberOfAPIs            = 1
+	expectedNumberOfEvents          = 2
+	expectedNumberOfTombstones      = 1
+	expectedNumberOfVendors         = 1
+
+	testTimeoutAdditionalBuffer = 1 * time.Minute
+)
+
+func TestORDAggregator(t *testing.T) {
+	appInput := createApp()
+
+	eventsMap := make(map[string]string, 0)
+	eventsMap[firstEventTitle] = firstEventDescription
+	eventsMap[secondEventTitle] = secondEventDescription
+
+	ctx := context.Background()
+
+	dexToken, err := idtokenprovider.GetDexToken()
+	require.NoError(t, err)
+
+	dexGraphQLClient := gql.NewAuthorizedGraphQLClient(dexToken)
+
+	app, err := fixtures.RegisterApplicationFromInput(t, ctx, dexGraphQLClient, testConfig.DefaultTenant, appInput)
+	require.NoError(t, err)
+
+	defer fixtures.UnregisterApplication(t, ctx, dexGraphQLClient, testConfig.DefaultTenant, app.ID)
+
+	t.Log("Create integration system")
+	intSys := fixtures.RegisterIntegrationSystem(t, ctx, dexGraphQLClient, "", "test-int-system")
+	require.NotEmpty(t, intSys)
+	defer fixtures.UnregisterIntegrationSystem(t, ctx, dexGraphQLClient, "", intSys.ID)
+
+	intSystemCredentials := fixtures.RequestClientCredentialsForIntegrationSystem(t, ctx, dexGraphQLClient, "", intSys.ID)
+	defer fixtures.DeleteSystemAuthForIntegrationSystem(t, ctx, dexGraphQLClient, intSystemCredentials.ID)
+
+	unsecuredHttpClient := http.DefaultClient
+	unsecuredHttpClient.Transport = &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	}
+
+	oauthCredentialData, ok := intSystemCredentials.Auth.Credential.(*directorSchema.OAuthCredentialData)
+	require.True(t, ok)
+
+	conf := &clientcredentials.Config{
+		ClientID:     oauthCredentialData.ClientID,
+		ClientSecret: oauthCredentialData.ClientSecret,
+		TokenURL:     oauthCredentialData.URL,
+	}
+
+	ctx = context.WithValue(ctx, oauth2.HTTPClient, unsecuredHttpClient)
+	httpClient := conf.Client(ctx)
+	httpClient.Timeout = 10 * time.Second
+
+	scheduleTime, err := parseCronTime(testConfig.AggregatorSchedule)
+	require.NoError(t, err)
+
+	defaultTestTimeout := scheduleTime + testTimeoutAdditionalBuffer
+	defaultCheckInterval := scheduleTime / 20
+
+	t.Run("Verifying ORD Document to be valid", func(t *testing.T) {
+		err = verifyORDDocument(defaultCheckInterval, defaultTestTimeout, func() bool {
+			var respBody string
+
+			// Verify system instances
+			respBody = makeRequestWithHeaders(t, httpClient, testConfig.ORDServiceURL+"/systemInstances?$format=json", map[string][]string{tenantHeader: {testConfig.DefaultTenant}})
+			if len(gjson.Get(respBody, "value").Array()) == 0 {
+				t.Log("Missing System Instances...will try again")
+				return false
+			}
+
+			require.Equal(t, expectedNumberOfSystemInstances, len(gjson.Get(respBody, "value").Array()))
+			require.Equal(t, applicationName, gjson.Get(respBody, "value.0.title").String())
+			require.Equal(t, applicationDescription, gjson.Get(respBody, "value.0.description").String())
+
```

Comment:
can we extract these lines into function and place it into pkg/assertions

Reply:
Makes sense, I guess it got a bit cleaner with this approach.